datamade/metro-pdf-merger
# Update logging strategy to capture messages from both app and worker

## Description

This PR updates our logging approach to capture app errors through Sentry's Flask integration and worker errors through Sentry's logging integration.

To do this, it:

- Initializes Sentry in the settings file, which is imported from by both the app and worker
- Adds the Sentry handler to the tasks logging config, conditional on whether a DSN has been configured
- Replaces `capture_exception` with `logger.exception`, so errors are included in the worker logs, as well as in Sentry

Handles #17, handles #28

### Testing instructions

I added the following block to `app.py` and [confirmed I received a message in Sentry](https://sentry.io/organizations/datamade/issues/2679607561/?project=155211&query=is%3Aunresolved):

```python
@app.route('/pong/')
def pong():

    raise ValueError('An unhandled app error occurred')

    try:
        from deployment import DEPLOYMENT_ID
    except ImportError:
        abort(401)

    return DEPLOYMENT_ID
```

I also added the following block to `tasks.py` and [confirmed I received a message with a breadcrumb in Sentry](https://sentry.io/organizations/datamade/issues/2679559837/?project=155211&query=is%3Aunresolved):

```python
def queue_daemon():
    logger.info('Starting worker')

    try:
        raise ValueError
    except:
        logger.exception('An exception occurred')

        import sys
        sys.exit()
```

Finally, I don't have Redis running locally, so I was also able to test that [unhandled worker errors appear as expected in Sentry](https://sentry.io/organizations/datamade/issues/2671093354/?project=155211&query=is%3Aunresolved).
